zarr/tests/test_core.py

```diff
@@ -2051,75 +2025,46 @@ def expected(self):
 
 @pytest.mark.skipif(have_fsspec is False, reason="needs fsspec")
 class TestArrayWithN5FSStore(TestArrayWithN5Store):
-
-    @staticmethod
-    def create_array(read_only=False, **kwargs):
+    def create_store(self):
         path = mkdtemp()
         atexit.register(shutil.rmtree, path)
         store = N5FSStore(path)
-        cache_metadata = kwargs.pop('cache_metadata', True)
-        cache_attrs = kwargs.pop('cache_attrs', True)
-        kwargs.setdefault('compressor', Zlib(1))
-        write_empty_chunks = kwargs.pop('write_empty_chunks', True)
-        init_array(store, **kwargs)
-        return Array(store, read_only=read_only, cache_metadata=cache_metadata,
-                     cache_attrs=cache_attrs, write_empty_chunks=write_empty_chunks)
+        return store
 
 
 class TestArrayWithDBMStore(TestArray):
 
-    @staticmethod
-    def create_array(read_only=False, **kwargs):
-        path = mktemp(suffix='.anydbm')
-        atexit.register(atexit_rmglob, path + '*')
-        store = DBMStore(path, flag='n')
-        cache_metadata = kwargs.pop('cache_metadata', True)
-        cache_attrs = kwargs.pop('cache_attrs', True)
-        write_empty_chunks = kwargs.pop('write_empty_chunks', True)
-        kwargs.setdefault('compressor', Zlib(1))
-        init_array(store, **kwargs)
-        return Array(store, read_only=read_only, cache_attrs=cache_attrs,
-                     cache_metadata=cache_metadata, write_empty_chunks=write_empty_chunks)
+    def create_store(self):
+        path = mktemp(suffix=".anydbm")
+        atexit.register(atexit_rmglob, path + "*")
```

Comment:
It would really be best to replace all of this stuff with [pytest tempdir fixtures](https://docs.pytest.org/en/6.2.x/tmpdir.html).

Reply:
yeah, I had the same feeling. I wanted to keep the initial PR close to the original tests, but I'm experimenting with a fixture-style locally.